src/pages/index.js

```diff
@@ -33,155 +36,168 @@ const initialCards = [
-const popupCaption = document.querySelector(".modal__caption");
-
-//validation
-const validationSettings = {
-  formSelector: ".modal__form",
-  inputSelector: ".modal__input",
-  submitButtonSelector: ".modal__button",
-  inactiveButtonClass: "modal__button_disabled",
-  inputErrorClass: "modal__input_type_error",
-  errorClass: "modal__error_visible",
-};
+const avatarUrlInput = avatarUpdateForm.querySelector("#avatar-link-input");
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "298dccb5-56b1-425a-ba81-960441ee84bf",
+    "Content-Type": "application/json",
+  },
+});
+
+const avatarCardPopUp = new PopUpWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
+);
+
+avatarCardPopUp.setEventListeners();
 
 const editFormValidator = new FormValidator(
   validationSettings,
   profileEditForm
 );
 const addFormValidator = new FormValidator(validationSettings, addCardForm);
+const updateAvatarValidator = new FormValidator(
+  validationSettings,
+  avatarUpdateForm
+);
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
 editFormValidator.disableButton();
 addFormValidator.disableButton();
+updateAvatarValidator.enableValidation();
+updateAvatarValidator.disableButton();
 
 /* const closeModal = (modalWindow) => {
   modalWindow.classList.remove("modal_open");
   document.removeEventListener("keydown", handleEscapeKey);
 }; */
 
 function createCard(item) {
-  const cardElement = new Card(item, "#card-template", handleCardClick);
+  const cardElement = new Card(
+    item,
+    "#card-template",
+    handleCardClick,
+    deleteCardSubmit,
+    handleLikeClick
+  );
   return cardElement.getView();
 }
 
-/* con = (data, wrap) => {
-  const card = createCard(data);
-  wrap.prepend(card);
-}; */
-
 const profileName = document.querySelector(".profile__name");
 const profileDescription = document.querySelector(".profile__description");
-
+const avatar = document.querySelector(".profile__image");
 const profileNameInput = document.querySelector("#profile-name-input");
 const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
 
-const userData = new UserInfo(profileName, profileDescription);
+function handleProfileEditSubmit(userProfile) {
+  popupWithEditForm.renderLoading(true);
+
+  api
+    .updateUserInfo(userProfile.name, userProfile.link)
+    .then(() => {
+      userData.setUserInfo(userProfile.name, userProfile.link);
+    })
+    .then(() => {
+      editFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      popupWithEditForm.renderLoading(false);
+    });
 
-function handleProfileEditSubmit() {
-  userData.setUserInfo(profileNameInput.value, profileDescriptionInput.value);
   popupWithEditForm.close();
 }
 
+function handleAvatarSubmit(data) {
+  avatarCardPopUp.renderLoading(true);
+
+  api
+    .updateUserAvatar(avatarUrlInput.value)
```

Comment:
`avatarUrlInput.value`

Only method `_getInputValues` collects all the data of the inputs **inside** `PopupWithForm` and passes it to the call of `submitForm`. So you need to use the collected data rather than the inputs.